packages/voting/src/deposit.rs

```diff
@@ -61,20 +73,33 @@ impl DepositInfo {
                     voting_module,
                     &cw_core_interface::voting::Query::TokenContract {},
                 )?;
-                token_addr
+                Ok(AssetInfo::Cw20(token_addr))
             }
-        };
-        // Make an info query as a smoke test that we are indeed
-        // working with a token here. We can't turbofish this
-        // type. See <https://github.com/rust-lang/rust/issues/83701>
-        let _info: cw20::TokenInfoResponse = deps
-            .querier
-            .query_wasm_smart(token.clone(), &cw20::Cw20QueryMsg::TokenInfo {})?;
-        Ok(CheckedDepositInfo {
-            token,
-            deposit,
-            refund_failed_proposals,
-        })
+        }?;
+
+        match token {
+            AssetInfoBase::Native(ref _denom) => Ok(CheckedDepositInfo {
+                token,
+                deposit,
+                refund_policy,
+            }),
+            AssetInfoBase::Cw20(ref addr) => {
+                // Make an info query as a smoke test that we are indeed
+                // working with a token here. We can't turbofish this
+                // type. See <https://github.com/rust-lang/rust/issues/83701>
+                let _info: cw20::TokenInfoResponse = deps
+                    .querier
+                    .query_wasm_smart(addr, &cw20::Cw20QueryMsg::TokenInfo {})?;
+                Ok(CheckedDepositInfo {
+                    token,
+                    deposit,
+                    refund_policy,
+                })
+            }
+            _ => Err(StdError::GenericErr {
+                msg: String::from("Unsupported asset"),
+            }),
```

Comment:
Will want to add a custom error here if we choose to merge this before doing the refactor.

Reply:
Yeah, I was lazy.